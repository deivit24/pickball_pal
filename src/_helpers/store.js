import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunk from 'redux-thunk';
import rootReducer from '../_reducers/root';

//Creating the store
const store = createStore(
  rootReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

export default store;
