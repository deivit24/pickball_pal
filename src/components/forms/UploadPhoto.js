// React
import React, { useState } from 'react';
// Redux
import { useDispatch } from 'react-redux';
// React Router DOM
import { useHistory } from 'react-router-dom';
// Upload Photo Action
import { uploadPhoto } from '../../_actions/profile';

import Button from '@material-ui/core/Button';

import '../../assets/css/UploadPhoto.css';

const UploadPhoto = () => {
  const history = useHistory();
  const dispatch = useDispatch();
  const [values, setValues] = useState({
    image: null,
    imageAlt: 'avatar',
  });

  const { image } = values;
  let disabled;
  if (image !== null) {
    disabled = false;
  } else {
    disabled = true;
  }
  const handleChange = (e) => {
    e.preventDefault();
    setValues({ image: e.target.files[0] });
  };
  const handleSubmit = (e) => {
    e.preventDefault();
    // get the first input element with the type of file
    const fd = new FormData();
    fd.append('image', image, image.name);

    dispatch(uploadPhoto(fd, history));
  };
  return (
    <div id="UploadPhoto" className="container">
      <div className="form-box">
        <div className="form-title">
          <h2>Upload Photo</h2>
        </div>
        <form
          noValidate
          autoComplete="on"
          encType="multipart/form-data"
          onSubmit={handleSubmit}
        >
          <div className="row">
            <div className="col-12">
              <input
                accept="image/*"
                type="file"
                label="Name"
                name="image"
                id="upload-photo"
                fullWidth
                margin="normal"
                variant="outlined"
                onChange={handleChange}
              />
            </div>
            <label htmlFor="upload-photo">
              <Button
                onClick={handleSubmit}
                variant="contained"
                color="primary"
                className="submit-button"
                disabled={disabled}
              >
                Upload
              </Button>
            </label>
          </div>
        </form>
      </div>
    </div>
  );
};

export default UploadPhoto;
