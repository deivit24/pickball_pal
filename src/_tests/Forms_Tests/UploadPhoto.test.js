import React from 'react';
import { render } from '@testing-library/react';
import UploadPhoto from '../../components/forms/UploadPhoto';
import { MemoryRouter } from 'react-router';
import { Provider } from 'react-redux';
import store from '../../_helpers/store';

//Testing UploadPhoto

describe('Testing the UploadPhoto', () => {
  it('it renders UploadPhoto without crashing', () => {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <UploadPhoto />
        </Provider>
      </MemoryRouter>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <UploadPhoto />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
