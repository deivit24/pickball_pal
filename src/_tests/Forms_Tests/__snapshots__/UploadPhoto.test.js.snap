// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the UploadPhoto matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
    id="UploadPhoto"
  >
    <div
      class="form-box"
    >
      <div
        class="form-title"
      >
        <h2>
          Upload Photo
        </h2>
      </div>
      <form
        autocomplete="on"
        enctype="multipart/form-data"
        novalidate=""
      >
        <div
          class="row"
        >
          <div
            class="col-12"
          >
            <input
              accept="image/*"
              id="upload-photo"
              label="Name"
              margin="normal"
              name="image"
              type="file"
              variant="outlined"
            />
          </div>
          <label
            for="upload-photo"
          >
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained submit-button MuiButton-containedPrimary Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              type="button"
            >
              <span
                class="MuiButton-label"
              >
                Upload
              </span>
            </button>
          </label>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
